Angular cli Generates Wrong Directory Path in app.module.ts

### Bug Report or Feature Request (mark with an `x`)
- [x] bug report
- [ ] feature request

### Versions.

node -v : v6.10.0
npm -v : 5.3.0
ng -v : 
```
@angular/cli: 1.2.4
node: 6.10.0
os: darwin x64
```

### Repro steps.

- install the latest angular-cli (version : 1.2.4)
- go in the desired repository
- run the command `ng new myProject`
- go in the created project
- go in myProject/src/app and create a folder "common"
- go in "common" folder 
- run `ng g c myComponent`


### The log given by the failure.
- the component is created at the right location : 
```
installing component
  create src/app/common/my-component/my-component.component.css
  create src/app/common/my-component/my-component.component.html
  create src/app/common/my-component/my-component.component.spec.ts
  create src/app/common/my-component/my-component.component.ts
  update src/app/app.module.ts
```
- but the path written in "app.module.ts" is wrong
```
ERROR in /Users/u.name/myProject/src/app/app.module.ts (5,38): Cannot find module './common/src/app/common/my-component/my-component.component'.

ERROR in ./src/app/app.module.ts
Module not found: Error: Can't resolve './common/src/app/common/my-component/my-component.component' in '/Users/u.name/myPoject/src/app'
 @ ./src/app/app.module.ts 10:0-99
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```


### Desired functionality.
The path written is constantly adding "src/app" part whereas older version of the cli did not have this behavior
The correct import statement should be : 
`import { MyComponentComponent } from './common/my-component/my-component.component';`


### Mention any other details that might be useful.
This issue has also been reported on StackOverflow by another user : 
https://stackoverflow.com/questions/45293787/angular-cli-generates-wrong-directory-path-in-app-module-ts

--- a/tests/e2e/tests/generate/component/component-module.ts
+++ b/tests/e2e/tests/generate/component/component-module.ts
@@ -1,14 +1,23 @@
+import * as fs from 'fs-extra';
 import {join} from 'path';
 import {ng} from '../../../utils/process';
 import {expectFileToMatch} from '../../../utils/fs';
 
 
 export default function() {
-  const modulePath = join('src', 'app', 'app.module.ts');
+  const root = process.cwd();
+  const modulePath = join(root, 'src', 'app', 'app.module.ts');
+
+  fs.mkdirSync('./src/app/sub-dir');
 
   return ng('generate', 'component', 'test-component', '--module', 'app.module.ts')
     .then(() => expectFileToMatch(modulePath,
       /import { TestComponentComponent } from '.\/test-component\/test-component.component'/))
+
+    .then(() => process.chdir(join(root, 'src', 'app')))
+    .then(() => ng('generate', 'component', 'test-component2', '--module', 'app.module.ts'))
+    .then(() => expectFileToMatch(modulePath,
+      /import { TestComponent2Component } from '.\/test-component2\/test-component2.component'/))
 
     // Try to run the unit tests.
     .then(() => ng('build'));

--- a/tests/e2e/tests/generate/directive/directive-module.ts
+++ b/tests/e2e/tests/generate/directive/directive-module.ts
@@ -1,14 +1,23 @@
+import * as fs from 'fs-extra';
 import {join} from 'path';
 import {ng} from '../../../utils/process';
 import {expectFileToMatch} from '../../../utils/fs';
 
 
 export default function() {
-  const modulePath = join('src', 'app', 'app.module.ts');
+  const root = process.cwd();
+  const modulePath = join(root, 'src', 'app', 'app.module.ts');
+
+  fs.mkdirSync('./src/app/sub-dir');
 
   return ng('generate', 'directive', 'test-directive', '--module', 'app.module.ts')
     .then(() => expectFileToMatch(modulePath,
       /import { TestDirectiveDirective } from '.\/test-directive.directive'/))
+
+    .then(() => process.chdir(join(root, 'src', 'app')))
+    .then(() => ng('generate', 'directive', 'test-directive2', '--module', 'app.module.ts'))
+    .then(() => expectFileToMatch(modulePath,
+      /import { TestDirective2Directive } from '.\/test-directive2.directive'/))
 
     // Try to run the unit tests.
     .then(() => ng('build'));

--- a/tests/e2e/tests/generate/guard/guard-module.ts
+++ b/tests/e2e/tests/generate/guard/guard-module.ts
@@ -1,15 +1,25 @@
+import * as fs from 'fs-extra';
 import {join} from 'path';
 import {ng} from '../../../utils/process';
 import {expectFileToMatch} from '../../../utils/fs';
 
 
 export default function() {
-  const modulePath = join('src', 'app', 'app.module.ts');
+  const root = process.cwd();
+  const modulePath = join(root, 'src', 'app', 'app.module.ts');
+
+  fs.mkdirSync('./src/app/sub-dir');
 
   return ng('generate', 'guard', 'test-guard', '--module', 'app.module.ts')
     .then(() => expectFileToMatch(modulePath,
       /import { TestGuardGuard } from '.\/test-guard.guard'/))
     .then(() => expectFileToMatch(modulePath,
       /providers:\s*\[TestGuardGuard\]/m))
+
+    .then(() => process.chdir(join(root, 'src', 'app')))
+    .then(() => ng('generate', 'guard', 'test-guard2', '--module', 'app.module.ts'))
+    .then(() => expectFileToMatch(modulePath,
+      /import { TestGuard2Guard } from '.\/test-guard2.guard'/))
+
     .then(() => ng('build'));
 }

--- a/tests/e2e/tests/generate/module/module-import.ts
+++ b/tests/e2e/tests/generate/module/module-import.ts
@@ -1,11 +1,15 @@
+import * as fs from 'fs-extra';
 import { join } from 'path';
 import { ng } from '../../../utils/process';
 import { expectFileToMatch } from '../../../utils/fs';
 
 export default function () {
-  const modulePath = join('src', 'app', 'app.module.ts');
+  const root = process.cwd();
+  const modulePath = join(root, 'src', 'app', 'app.module.ts');
   const subModulePath = join('src', 'app', 'sub', 'sub.module.ts');
   const deepSubModulePath = join('src', 'app', 'sub', 'deep', 'deep.module.ts');
+
+  fs.mkdirSync('./src/app/sub-dir');
 
   return Promise.resolve()
     .then(() => ng('generate', 'module', 'sub'))
@@ -42,4 +46,10 @@ export default function () {
     .then(() => expectFileToMatch(deepSubModulePath,
       /import { Test6Module } from '..\/..\/test6\/test6.module'/))
     .then(() => expectFileToMatch(deepSubModulePath, /imports: \[(.|\s)*Test6Module(.|\s)*\]/m)));
+
+    .then(() => process.chdir(join(root, 'src', 'app')))
+    .then(() => ng('generate', 'module', 'test7', '--module', 'app.module.ts')))
+    .then(() => expectFileToMatch(modulePath,
+      /import { Test7Module } from '.\/test7\/test7.module'/))
+    .then(() => expectFileToMatch(modulePath, /imports: \[(.|\s)*Test7Module(.|\s)*\]/m))
 }

--- a/tests/e2e/tests/generate/pipe/pipe-module.ts
+++ b/tests/e2e/tests/generate/pipe/pipe-module.ts
@@ -1,14 +1,23 @@
+import * as fs from 'fs-extra';
 import {join} from 'path';
 import {ng} from '../../../utils/process';
 import {expectFileToMatch} from '../../../utils/fs';
 
 
 export default function() {
-  const modulePath = join('src', 'app', 'app.module.ts');
+  const root = process.cwd();
+  const modulePath = join(root, 'src', 'app', 'app.module.ts');
+
+  fs.mkdirSync('./src/app/sub-dir');
 
   return ng('generate', 'pipe', 'test-pipe', '--module', 'app.module.ts')
     .then(() => expectFileToMatch(modulePath,
       /import { TestPipePipe } from '.\/test-pipe.pipe'/))
+
+    .then(() => process.chdir(join(root, 'src', 'app')))
+    .then(() => ng('generate', 'pipe', 'test-pipe2', '--module', 'app.module.ts'))
+    .then(() => expectFileToMatch(modulePath,
+      /import { TestPipe2Pipe } from '.\/test-pipe2.pipe'/))
 
     // Try to run the unit tests.
     .then(() => ng('build'));

--- a/tests/e2e/tests/generate/service/service-module.ts
+++ b/tests/e2e/tests/generate/service/service-module.ts
@@ -1,14 +1,23 @@
+import * as fs from 'fs-extra';
 import {join} from 'path';
 import {ng} from '../../../utils/process';
 import {expectFileToMatch} from '../../../utils/fs';
 
 
 export default function() {
-  const modulePath = join('src', 'app', 'app.module.ts');
+  const root = process.cwd();
+  const modulePath = join(root, 'src', 'app', 'app.module.ts');
+
+  fs.mkdirSync('./src/app/sub-dir');
 
   return ng('generate', 'service', 'test-service', '--module', 'app.module.ts')
     .then(() => expectFileToMatch(modulePath,
       /import { TestServiceService } from '.\/test-service.service'/))
+
+    .then(() => process.chdir(join(root, 'src', 'app')))
+    .then(() => ng('generate', 'service', 'test-service2', '--module', 'app.module.ts'))
+    .then(() => expectFileToMatch(modulePath,
+      /import { TestService2Service } from '.\/test-service2.service'/))
 
     // Try to run the unit tests.
     .then(() => ng('build'));

